napari/utils/progress.py

```diff
@@ -0,0 +1,245 @@
+import inspect
+from contextvars import ContextVar
+from typing import Iterable, Optional
+
+from qtpy import QtCore
+from qtpy.QtWidgets import (
+    QApplication,
+    QHBoxLayout,
+    QLabel,
+    QProgressBar,
+    QVBoxLayout,
+    QWidget,
+)
+from tqdm import tqdm
+
+from .._qt.utils import get_viewer_instance
+
+IS_NESTED = ContextVar('IS_NESTED', default=False)
+
+
+def get_pbar(viewer_instance, **kwargs):
+    """Adds ProgressBar to viewer Activity Dock and returns it.
+
+    Parameters
+    ----------
+    viewer_instance : qtViewer
+        current napari qtViewer instance
+
+    Returns
+    -------
+    ProgressBar
+        progress bar to associate with current iterable
+    """
+    pbar = ProgressBar(**kwargs)
+    pbr_layout = viewer_instance.activityDock.widget().layout
+    if IS_NESTED.get():
+        last_added_idx = pbr_layout.count() - 1
+        unnested_widg_layout = (
+            pbr_layout.itemAt(last_added_idx).widget().layout()
+        )
+        unnested_widg_layout.addWidget(pbar)
```

Comment:
it would also be nice if there was a bit more visual queue that these were actually grouped progress bars (not just adjacent).  Like a QFrame with a border, or a shaded background, etc...

Reply:
yeah I was going to put this in the styling PR alongside all the other styling things, especially since now we've moved nesting out of this PR as well...